src/main/java/uk/gov/pay/adminusers/service/InviteCreator.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.pay.adminusers.service;
+
+import com.google.inject.persist.Transactional;
+import org.slf4j.Logger;
+import uk.gov.pay.adminusers.app.config.LinksConfig;
+import uk.gov.pay.adminusers.logger.PayLoggerFactory;
+import uk.gov.pay.adminusers.model.Invite;
+import uk.gov.pay.adminusers.model.InviteServiceRequest;
+import uk.gov.pay.adminusers.persistence.dao.InviteDao;
+import uk.gov.pay.adminusers.persistence.dao.RoleDao;
+import uk.gov.pay.adminusers.persistence.dao.UserDao;
+import uk.gov.pay.adminusers.persistence.entity.InviteEntity;
+import uk.gov.pay.adminusers.persistence.entity.UserEntity;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.UriBuilder;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static javax.ws.rs.core.UriBuilder.fromUri;
+import static uk.gov.pay.adminusers.app.util.RandomIdGenerator.randomUuid;
+import static uk.gov.pay.adminusers.model.InviteType.SERVICE;
+import static uk.gov.pay.adminusers.service.AdminUsersExceptions.*;
+
+public class InviteCreator {
+
+    private static final Logger LOGGER = PayLoggerFactory.getLogger(InviteCreator.class);
+    private static final String SELFSERVICE_INVITES_PATH = "invites";
+    private static final String SELFSERVICE_LOGIN_PATH = "login";
+    private static final String SELFSERVICE_FORGOTTEN_PASSWORD_PATH = "reset-password";
+    private static final String FEEDBACK_PATH = "support.html";
```

Comment:
We should consider moving paths into config, _Adminusers_ is knowing too much about where these resources outside are living. No for this PR but something we can discuss if agreed.